rest/client/rest.go

```diff
@@ -20,6 +20,27 @@ import (
 	"github.com/pkg/errors"
 )
 
+// respErrorf attempts to read a gimlet.ErrorResponse from the response body
+// JSON. If successful, it returns the gimlet.ErrorResponse wrapped with the
+// HTTP status code and the formatted error message. Otherwise, it returns an
+// error message with the HTTP status and raw response body.
+func respErrorf(resp *http.Response, format string, args ...interface{}) error {
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		err = errors.Wrapf(err, "status code %d", resp.StatusCode)
+		return errors.Wrap(err, "could not read response body")
```

Comment:
Should this also wrap with the passed in format string? If so, you might want to combine the errors that are already here

Reply:
Done. I wrapped all the return cases with the formatted error context.